src/xdp/trn_transit_xdp.c

```diff
@@ -460,8 +460,50 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 				return XDP_ABORTED;
 			}
 		} else {
+			pkt->pod_label_value_opt = (void *)pkt->scaled_ep_opt + sizeof(*pkt->scaled_ep_opt);
```

Comment:
seems like you are forcing tracked_state to be null here for label based network policy. How do you know if the environment is label based or ip based? Consider separate this portion of the code from ip based network policy

Reply:
There is only one environment. In the environment, when sending out the traffic, the code will always add label values to geneve header. And when receiving the traffic (the code here), the code will always retrieve the label values from geneve header.
The code here is in the function trn_process_inner_ip, which is contains by function trn_process_inner_eth, which is contains by function trn_process_geneve. The code is in "controlled" environment which rules the traffic must be from another side which is controlled by mizar. The both sides are implemented by mizar to use geneve to carry on additional flags for the traffic.